app/services/route_summary_card_data_service.rb

```diff
@@ -0,0 +1,84 @@
+class RouteSummaryCardDataService
+  attr_reader :page, :pages
+
+  class << self
+    def call(**args)
+      new(**args)
+    end
+  end
+
+  def initialize(page:, pages:)
+    @page = page
+    @pages = pages
+  end
+
+  def build_data
```

Comment:
What about calling this `summary_card_data`?

```suggestion
  def summary_card_data
```